src/eos-updater-poll-common.c

```diff
@@ -840,6 +840,44 @@ get_origin_refspec (OstreeDeployment *booted_deployment,
 }
 
 static void
+get_custom_hw_descriptors (GHashTable *hw_descriptors,
+                           const gchar *path)
+{
+  g_autoptr(GKeyFile) keyfile = NULL;
+  g_auto(GStrv) keys = NULL;
+  gchar **iter;
+  const gchar *group = "descriptors";
+
+  keyfile = g_key_file_new ();
+  if (!g_key_file_load_from_file (keyfile,
+                                  path,
+                                  G_KEY_FILE_NONE,
+                                  NULL))
+    return;
+
+  keys = g_key_file_get_keys (keyfile,
+                              group,
+                              NULL,
+                              NULL);
+  if (keys == NULL)
+    return;
+
+  for (iter = keys; *iter != NULL; ++iter)
+    {
+      const gchar *key = *iter;
+      gchar *value = g_key_file_get_string (keyfile,
+                                            group,
+                                            key,
+                                            NULL);
+
+      if (value == NULL)
+        continue;
+
+      g_hash_table_insert (hw_descriptors, g_strdup (key), value);
+    }
```

Comment:
Surprising that `GKeyFile` doesn't have this functionality. Maybe it would be worth generalising this into `hash_table_for_keyfile_group` and getting the `descriptors` group separately.